### DISCUSSION TITLE
Can checkPermissions be public instead of private?

### DISCUSSION THREAD

**Discussion by lmillett** [AsyncSwift/AsyncLocationKit#17] [Ideas]
In my app I would like to check what location permissions have already been authorized before I ask them to grant me location permission. That way I'm not forced to ask for permissions before they have even used the app. I can delay it to a later point when they do an operation that really requires that authorization.

I have that change made locally to my package and it seems to work as I would like it to.

Happy to submit that change if others agree.

**Comment by gre4ixin** [comment]
https://github.com/AsyncSwift/AsyncLocationKit/pull/19

**Comment by lmillett** [comment]
Thanks for the update!

**Reply by gre4ixin** [reply]
You're welcome)
